### ISSUE TITLE
Notifications don't live through a redirect

### ISSUE THREAD

**Issue by bleroy** [bleroy/OrchardImportTest#325]
Take the comment create action for example. When a new comment doesn't
immediately appear for some reason (e.g. pending by default) a message is
stuffed into the notifier and the request is redirected to the return URL.
Upon reaching the destination (return URL) no message is displayed since the
message is lost when the NotifyFilter bails in OnResultExecuting if the result
is not a ViewResultBase (as is the case in a redirect). There should be
something in place to persist the notification (all relevant messages, really)
in this situation.  

Work Item Details
--------------------
**Original CodePlex Issue:**	[Issue 16435](http://orchard.codeplex.com/workitem/16435)
**Status:**	Active
**Reason Closed:**	Unassigned
**Assigned to:**	Unassigned
**Reported on:**	Jun 1, 2010 at 11:32 PM
**Reported by:**	skewed
**Updated on:**	Jul 5, 2012 at 6:44 PM
**Updated by:**	sebastienros
**Status:**	Active
**Reason Closed:**	Unassigned
**Assigned to:**	Unassigned
**Reported on:**	Jun 1, 2010 at 11:32 PM
**Reported by:**	skewed
**Updated on:**	Jul 5, 2012 at 6:44 PM
**Updated by:**	sebastienros

**Comment by bleroy** [comment]
On *2010-06-09 05:11:23 UTC*, **sebastienros** commented:

Actually it doesn't appear if the comment is posted on the Home Page, through the homepagecontroller. For instance, by going directly to /Home, the entering a comment, the message appears. Though the redirect is also called. Can reproduce, but I couldn't
 correct it.

**Comment by bleroy** [comment]
On *2012-07-06 01:43:55 UTC*, **sebastienros** commented:

The solution might be to save the messages in a cookie if the action result is not of ViewResultBase. This is postponed right now as too risky for the upcoming release.
